### PR TITLE
add 'skipMergeProdBranch' option to hotfix:finish (similar to skipMergeDevBranch)

### PR DESCRIPTION
**Pull request by chulan**
when creating hotfixes, especially to older versions, merging the change into master is not necessary.

### CODE REVIEW

File: src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java [hunk 1]
```diff
@@ -102,6 +102,14 @@
     @Parameter(property = "useSnapshotInHotfix", defaultValue = "false")
     private boolean useSnapshotInHotfix;
 
+
+    /**
+     * Whether to merge hotfix into master.
+     * 
+     */
+    @Parameter(property = "skipMergeMaster", defaultValue = "false")
+    private boolean skipMergeMaster = false;
```
Comment (aleksandr-m) [inline src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java]: It is better to name it `skipMergeProduction` or `skipMergeProdBranch` inline with `skipMergeDevBranch`.

File: src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java [hunk 2]
```diff
@@ -220,6 +219,20 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                         commitMessages.getTagHotfixMessage(), gpgSignTag, properties);
             }
 
+            if (supportBranchName != null) {
```
Comment (aleksandr-m) [inline src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java]: Why did you move merging after tagging? It will probably create tag on the wrong commit.

**Comment by glianeric** [comment]
Hi @aleksandr-m ,

  I asked @chulan to make this change, so I will explain if you don't mind :)

The history here is that we were using the jgitflow plugin which, as you probably know, has been abandoned so I'm looking to switch to your plugin.   jgitflow did have switches to avoid merging, and I'm looking to keep that workflow.   We do use support branches as well, but in our workflow they are the exception; hotfix branches are tagged and left behind (the branch is deleted, the tag remains) as software is quickly moved to newer versions of dependencies.

Thanks! :)

File: src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java (hunk 1 above)
Comment (chulan) [reply src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java]: agreed, I'll change it to  `skipMergeProdBranch`

File: src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java (hunk 2 above)
Comment (chulan) [reply src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java]: I want to tag the hotfix branch, and not the production branch which never gets the changes merged in our use case.  Moving the merge to after tagging does that, and in the "normal' use of merging to production the tag is still created.  From my testing the change results in the tag being associated with the last commit of the branch instead of with the merge commit. 

Another approach would be to revert to merging before tagging, but not checking out production when 'skipMergeProdBranch' is specified. Tagging would then occur as normal.
Then immediately after the tagging check for the 'skipMergeProdBranch' (and supportBranchName is null) to ensure the expected branch is active for subsequent processing:
```suggestion
             if (skipMergeProdBranch && (supportBranchName == null)) {
                // git checkout master 
                gitCheckout(gitFlowConfig.getProductionBranch());
                }
```

Appreciate your insight

Comment (aleksandr-m) [reply src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java]: Do you need to checkout production branch at all if `skipMergeProdBranch` is enabled?

Comment (chulan) [reply src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java]: We need to switch off the hotfix branch so it can be deleted.  I think switching to production is safest as it aligns with the existing logic

Comment (aleksandr-m) [reply src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java]: Yeah, it is seems that `if (skipMergeProdBranch && (supportBranchName == null))` is better than other alternatives where it is also needed to adjust currentVersion.

Go ahead!

Comment (aleksandr-m) [reply src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java]: Can you also squash all the commits, after resolving this.

Comment (chulan) [reply src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java]: will do, thanks

Comment (chulan) [reply src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java]: Just pushed the latest.  Let me know if anything needs adjustment

**Comment by aleksandr-m** [comment]
@chulan Good work. Thank you.

**Comment by aleksandr-m** [comment]
@chulan @glianeric 1.12.0 is out.

**Comment by glianeric** [comment]
Excellent, thank you! :)

**Comment by chulan** [comment]
Great, thanks!
